FatPkg/EnhancedFatDxe/DiskCache.c

```diff
@@ -8,6 +8,229 @@ SPDX-License-Identifier: BSD-2-Clause-Patent
 
 #include "Fat.h"
 
+/**
+  Helper function to check if any LBA is dirty in this cache line.
+
+  @param[in]    CacheTag   - CacheTag to check
+
+  @retval       TRUE       - Cache is Dirty
+                FALSE      - Cache is not Dirty
+
+**/
+STATIC
+BOOLEAN
+IsCacheTagDirty (
+  IN CACHE_TAG  *CacheTag
+  )
+{
+  if (CacheTag == NULL) {
+    ASSERT (CacheTag != NULL);
+    return FALSE;
+  }
+
+  return CacheTag->Dirty;
+}
+
+/**
+  Helper function to clear the dirty state of the cache line.
+
+  @param[in]    CacheTag   - CacheTag to clear
+
+**/
+STATIC
+VOID
+ClearCacheTagDirtyState (
+  IN CACHE_TAG  *CacheTag
+  )
+{
+  if (CacheTag == NULL) {
+    ASSERT (CacheTag != NULL);
+    return;
+  }

```

Comment:
CacheTag should NOT be NULL. So how about just "ASSERT()"?

Reply:
I agree, CodeQL is not that smart and will flag it if we don’t check, so leaving the check in.